test/unit/components/join-flow-step.test.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {mountWithIntl} from '../../helpers/intl-helpers.jsx';
+import JoinFlowStep from '../../../src/components/join-flow/join-flow-step';
+
+describe('JoinFlowStep', () => {
+
+    test('test components exist when props present', () => {
+        const props = {
+            children: null,
+            className: 'join-flow-step-class',
+            description: 'description text',
+            headerImgSrc: '/image.png',
+            onSubmit: jest.fn(),
+            nextButton: 'some stuff',
+            title: 'join flow step title',
+            waiting: true
+        };
+        const component = mountWithIntl(
+            <JoinFlowStep
+                {...props}
+            />
+        );
+        expect(component.find('div.join-flow-header-image').exists()).toEqual(true);
+        expect(component.find({src: props.headerImgSrc}).exists()).toEqual(true);
+
+        expect(component.find('.join-flow-inner-content').exists()).toEqual(true);
+        expect(component.find('.join-flow-title').exists()).toEqual(true);
+
+        expect(component.find('.join-flow-title').first()
+            .prop('title')).toEqual(props.title);
+
+        expect(component.find('div.join-flow-description').exists()).toEqual(true);
+        expect(component.find('div.join-flow-description').text()).toEqual(props.description);
+
+
+        component.find('button[type="submit"]').simulate('submit');
```

Comment:
I suggest moving this to a separate test -- I try (having learned this from others) to keep my tests as narrowly focused as possible, so that if a test fails, you immediately know the sort of thing that's going wrong

Reply:
Good call. Was thinking the setup was the same, but it isn't :)